ui/site/src/socket.js

```diff
@@ -65,14 +65,15 @@ lichess.StrongSocket = function(url, version, settings) {
   };
 
   var send = function(t, d, o, noRetry) {
-    d = d == null ? {} : d,
-      o = o || {};
-    if (o.withLag) d.l = Math.round(averageLag);
+    o = o || {};
+    var msg = { t: t };
+    if (d !== undefined) {
+      if (o.withLag) d.l = Math.round(averageLag);
+      if ('millis' in o) d.s = Math.floor(o.millis * 0.1).toString(36);
```

Comment:
it's not a very good practice to use `in` operator since it does look up all the prototype chain.
it's better to do strict comparisons: `if (o.millis !== undefined && o.millis !== null)` even if it's more verbose